File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    up_down = [min(ord(c) - 65, 91 - ord(c)) for c in name]
    min_dist = len(up_down) - 1
    answer = sum(up_down)
    
    for i in range(len(up_down)):
        next = i + 1
        while next < len(up_down) and up_down[next] == 0:
            next += 1
        
        min_dist = min(
            min_dist,
            i * 2 + len(up_down) - next,
            i + (len(up_down) - next) * 2
        )
    
    return answer + min_dist

File: 프로그래머스/2/42860. 조이스틱/조이스틱.js
function solution(name) {
    let answer = 0;
    let min_dist = name.length - 1;
    
    const up_down = name.split("").map(c => {
        const dist = Math.min(c.charCodeAt() - 65, 91 - c.charCodeAt());
        answer += dist;
        return dist;
    });
    
    for (let i = 0; i < up_down.length; i++) {
        let next = i + 1;
        
        while (next < up_down.length && up_down[next] === 0) {
            next++;
        }
        
        min_dist = Math.min(
            min_dist,
            i * 2 + up_down.length - next,
            i + (up_down.length - next) * 2
        );
    }
    
    return answer + min_dist;
}